compiler/luci-interpreter/src/kernels/Mean.cpp

```diff
@@ -0,0 +1,247 @@
+      if (current == previous)
+      {
+        --reduction_count;
+        break;
+      }
+    }
+  }
+  return reduction_count;
+}
+
+static void ResizeOutputTensor(const Shape &input_shape, int input_num_dims, const int *axis_data,
+                               int num_axis, bool keep_dims, Tensor *output)
+{
+  if (input_num_dims == 0)
+  {
+    output->resize(Shape(0));
+    return;
+  }
+
+  if (keep_dims)
+  {
+    Shape output_shape(input_num_dims);
+    for (int idx = 0; idx < input_num_dims; ++idx)
+    {
+      bool is_axis = false;
+      for (int axis_idx = 0; axis_idx < num_axis; ++axis_idx)
+      {
+        if (axis_data[axis_idx] == idx || axis_data[axis_idx] + input_num_dims == idx)
+        {
+          is_axis = true;
+          break;
+        }
+      }
+      if (is_axis)
+      {
+        output_shape.dim(idx) = 1;
+      }
+      else
+      {
+        output_shape.dim(idx) = input_shape.dim(idx);
+      }
+    }
+    output->resize(output_shape);
+  }
+  else
+  {
+    int num_reduce_axis = GetAxisReductionCount(axis_data, num_axis, input_num_dims);
+    Shape output_shape(input_num_dims - num_reduce_axis);
+    int num_skip_axis = 0;
+    for (int idx = 0; idx < input_num_dims; ++idx)
+    {
+      bool is_axis = false;
+      for (int axis_idx = 0; axis_idx < num_axis; ++axis_idx)
+      {
+        if (axis_data[axis_idx] == idx || axis_data[axis_idx] + input_num_dims == idx)
+        {
+          ++num_skip_axis;
+          is_axis = true;
+          break;
+        }
+      }
+      if (!is_axis)
+      {
+        output_shape.dim(idx - num_skip_axis) = input_shape.dim(idx);
+      }
+    }
+    output->resize(output_shape);
+  }
+}
+
+Mean::Mean(const Tensor *input, const Tensor *axis, Tensor *output, const ReducerParams &params)
+    : KernelWithParams<ReducerParams>(params), _input(input), _axis(axis), _output(output)
+{
+}
+
+void Mean::configure()
+{
+  assert(_input->element_type() == _output->element_type());
+  assert(_axis->element_type() == DataType::S32);
+  const Shape &input_shape = _input->shape();
+  int input_num_dims = input_shape.num_dims();
+
+  const auto *axis_data = getTensorData<int32_t>(_axis);
+  int num_axis = _axis->shape().num_elements();
+  assert(num_axis <= 4);
+
+  tflite::MeanParams params{};
+  ResolveAxis(axis_data, num_axis, &params);
+  const bool need_temporaries =
+      !(_params.keep_dims && input_num_dims == 4 && params.axis_count == 2 &&
+        ((params.axis[0] == 1 && params.axis[1] == 2) ||
+         (params.axis[0] == 2 && params.axis[1] == 1)));
+  if (need_temporaries)
+  {
+    _temp_index =
+        std::make_unique<Tensor>(DataType::S32, Shape(input_num_dims), AffineQuantization{}, "");
+    _resolved_axis =
+        std::make_unique<Tensor>(DataType::S32, Shape(num_axis), AffineQuantization{}, "");
+    _temp_sum = std::make_unique<Tensor>(
+        _input->element_type(), Shape(_output->shape().num_elements()), AffineQuantization{}, "");
```

Comment:
Output tensor is not resized yet (it is resized below), you can't use its shape here.
Also, this would be more clear:
```suggestion
        _input->element_type(), _output->shape(), AffineQuantization{}, "");
```

Reply:
Ok.